addons/web/models/models.py

```diff
@@ -66,7 +66,7 @@ def web_search_read(self, domain=None, fields=None, offset=0, limit=None, order=
                 'records': []
             }
         if limit and (len(records) == limit or self.env.context.get('force_search_count')):
-            length = self.search_count(domain)
+            length = self.search_count(domain, limit=10000)
```

Comment:
Personally I would really be in favour of adding an `ir.config_parameter` from which this gets the value.
This way more technical partners could fine-tune this number by switching the parameter to another value. There's customers who'd want this number to be much lower and some would like it to be higher :) The difference in performance would be very small while making it totally configurable.

Reply:
+1

Reply:
Or even just a const that another JS module the partner could make could update if needed